gostream/ffmpeg/avlog/avlog.go

```diff
@@ -0,0 +1,53 @@
+// Package avlog is libav's logging facilities
+// See https://ffmpeg.org/doxygen/3.0/log_8h.html
+package avlog
+
+//#cgo CFLAGS: -I${SRCDIR}/../include
+//#cgo linux,arm64 LDFLAGS: -L${SRCDIR}/../Linux-aarch64/lib -lavformat -lavcodec -lavutil
+//#cgo linux,amd64 LDFLAGS: -L${SRCDIR}/../Linux-x86_64/lib -lavformat -lavcodec -lavutil
+//#cgo linux,arm LDFLAGS: -L${SRCDIR}/../Linux-armv7l/lib -lavformat -lavcodec -lavutil
+//#include <libavutil/log.h>
+import "C"
+
+const (
```

Comment:
[q] It looks like we are missing a couple of macros here. Do you think we should add these?
- [AV_LOG_PRINT_LEVEL](https://ffmpeg.org/doxygen/3.0/group__lavu__log.html#ga8b447fa5f9d7e243be4634878ed1a0ca)   2
- [AV_LOG_SKIP_REPEATED](https://ffmpeg.org/doxygen/3.0/group__lavu__log.html#ga6cdf5cd331b17e80e8308124f05a6db8)   1

Reply:
[q] Is it possible and/or good practice to use macros directly from the the headerfile (cgo import), instead of defining our own data structures for them?

Reply:
I'm not using either of those, so I didn't include them. I only added the rest of the log levels so we wouldn't have to look them up later. I was also playing around with the different levels and decided to leave them in. I generally tried to avoid importing things that aren't tested or used in prod.

From what I've seen, importing macros from C isn't really a thing. People have been getting around it by defining a new C function that wraps the macro. That's how it's done in avutil.go, where `av_err2str` is a macro.
```
//#include <libavutil/error.h>
//#include <stdlib.h>
// static const char *error2string(int code) { return av_err2str(code); }
import "C"
...
// ErrorFromCode returns an error from the given code
func ErrorFromCode(code int) error {
	if code >= 0 {
		return nil
	}

	return errors.New(C.GoString(C.error2string(C.int(code))))
}
```
Because you can't call a function and use its value to instantiate a constant, I just added the values myself. But to answer your question, yes, I think it'd be better practice to use the macros directly somehow. I'm just not sure how at the moment. If I find out, or if you know of a way, I'd be happy to update this!